src/lib.rs

```diff
@@ -686,6 +663,60 @@ impl TestData {
 mod tests {
     use super::*;
 
+    #[test]
+    #[cfg(feature = "zeroize")]
+    fn test_zeroize() {
+        fn run() {
+            use crate::{keypair, CRYPTO_PUBLICKEYBYTES, CRYPTO_SECRETKEYBYTES};
+
+            let mut pk_buffer = [0u8; CRYPTO_PUBLICKEYBYTES];
+            let mut sk_buffer = [5u8; CRYPTO_SECRETKEYBYTES];
+            let mut rng = rand::thread_rng();
+
+            let zeroed_key = [0; CRYPTO_SECRETKEYBYTES];
+
+            let (_, secret_key) = keypair(&mut pk_buffer, &mut sk_buffer, &mut rng);
+            drop(secret_key);
+
+            assert_eq!(zeroed_key, sk_buffer);
+        }
+
+        std::thread::Builder::new()
+            // Use a large enough stack size to run all kem variants with the key buffers on the stack.
+            .stack_size(4 * 1024 * 1024)
+            .spawn(run)
+            .unwrap()
+            .join()
+            .unwrap();
+    }
+
+    #[test]
+    #[cfg(feature = "mceliece8192128f")]
+    fn test_crypto_kem_api_keypair() {
```

Comment:
hm I think for clarifications we should rename this into `test_crypto_alloc_api_keypair` since this tests the `keypair_boxed` functionality of the `alloc` feature. (in order to resolve confusion about kem)

There is this ambiguity between `kem` as in the 'feature' and KEM as in 'key encapsulation mechanism' which causes some trouble I guess

Reply:
Done. I also removed `test_` from the test names. I always find this redundant since they are always placed in a module named `tests` anyway. When you run them cargo will print `test tests::zeroize ... ok` anyway.

Reply:
awesome thank you :) (I hope this was not too nitpicky)